product_packaging_dimension/models/product_packaging.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2019-2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class ProductPackaging(models.Model):
+    _inherit = "product.packaging"
+    # Constraints and dimensional field names are identical
+    # to product_packaging.py in delivery.
+    # The redundancy here avoids unnecessary dependencies on sale modules.
+
+    _sql_constraints = [
+        ("positive_height", "CHECK(height>=0.0)", "Height must be positive"),
+        ("positive_width", "CHECK(width>=0.0)", "Width must be positive"),
+        ("positive_length", "CHECK(packaging_length>=0.0)", "Length must be positive"),
+        (
+            "positive_max_weight",
+            "CHECK(max_weight>=0.0)",
+            "Max Weight must be positive",
+        ),
+    ]
+    height = fields.Float("Height")
+    width = fields.Float("Width")
+    packaging_length = fields.Float("Length")
+
+    length_uom_id = fields.Many2one(
+        "uom.uom",
+        "Dimensions Units of Measure",
+        domain=lambda self: [
+            ("category_id", "=", self.env.ref("uom.uom_categ_length").id)
+        ],
+        help="UoM for packaging length, height, width (based on lenght UoM)",
+        default=lambda self: self.env[
+            "product.template"
+        ]._get_length_uom_id_from_ir_config_parameter(),
+    )
+    length_uom_name = fields.Char(
+        string="Length unit of measure label",
+        related="length_uom_id.name",
+        readonly=True,
+    )
+
+    weight = fields.Float("Weight")
```

Comment:
@Kev-Roche Could you highlight me why this field has been added during the migration to v14?
In a lot of modules of the wms repo as well as in the enterprise delivery_* modules, it is the standard max_weight field that is used. I find this second weight field very confusing since v14

cc @sebalix @simahawk @jgrandguillaume

Reply:
Indeed, seems not clear what is used for

Reply:
Having max_weight in this module already prevented unnecessary dependency to the delivery module

Reply:
Max_weight is used as the product packaging weight at those places:

* https://github.com/OCA/stock-logistics-warehouse/blob/14.0/stock_measuring_device/wizard/measuring_wizard.py#L85
* https://github.com/OCA/product-attribute/blob/14.0/product_total_weight_from_packaging/models/product.py#L36